torch/distributed/pipelining/_IR.py

```diff
@@ -875,6 +885,41 @@ def move_param_to_callee(
                     node.target,
                 )
 
+        # [aliasing] store tensor id -> list of FQNs, built from state dict
+        id_to_fqns: Dict[int, List[str]] = defaultdict(list)
+        for fqn, tensor in mod.state_dict(keep_vars=True).items():
+            id_to_fqns[id(tensor)].append(fqn)
+
+        # After moving the params to their corresponding hierarchies, we also
+        # need to move the `get_attr` nodes from the root of the graph to those
+        # hierarchies.
+        # [aliasing] use id -> fqn mapping to list out all valid FQNs
+        inputs_to_state: Dict[str, List[str]] = {}
+        for attr in attr_nodes:
+            tensor = _recursive_getattr(mod, attr.target)
+            inputs_to_state[attr.name] = [fqn for fqn in id_to_fqns[id(tensor)]]
+
+        # [aliasing] for each submodule split, assign attributes on FQNs that may be used.
+        # We determine this based on whether or not the FQN attribute parent exists.
+        # i.e. if the last submodule exists, assign the attribute.
+        added_attributes: Dict[str, List[str]] = defaultdict(list)
+        for fqn, tensor in mod.state_dict(keep_vars=True).items():
+            for name, submod in split.named_children():
+                if isinstance(submod, fx.GraphModule):
+                    if _recursive_getattr(submod, fqn) is None:
+                        # try to reach the last submodule, if it exists.
+                        atoms = fqn.split(".")
+                        has_parent = True
+                        mod_itr = submod
+                        for atom in atoms[:-1]:
+                            if not hasattr(mod_itr, atom):
+                                has_parent = False
+                                break
+                            mod_itr = getattr(mod_itr, atom)
```

Comment:
nit: to avoid doing two searches (one in `_recursive_getattr` the other in code here), we can ask `_recursive_getattr` to always return the last-matched module hierarchy (e.g. `mod_itr` here).
- If `fqn` is indeed under `submod`, then the return would be its true parent;
- Otherwise, the return would be the most common ancestor.
We can then compare `mod_itr.split(".")` with `fqn.split(".")[:-1]` to determine `has_parent`.